crates/subspace-networking/src/node_runner.rs

```diff
@@ -463,29 +479,54 @@ where
                     }
                 }
             }
-            SwarmEvent::OutgoingConnectionError { peer_id, error } => match error {
-                DialError::Transport(ref addresses) => {
-                    for (addr, _) in addresses {
-                        debug!(?error, ?peer_id, %addr, "SwarmEvent::OutgoingConnectionError (DialError::Transport) for peer.");
-                        if let Some(peer_id) = peer_id {
-                            self.networking_parameters_registry
-                                .remove_known_peer_addresses(peer_id, vec![addr.clone()])
-                                .await;
+            SwarmEvent::OutgoingConnectionError { peer_id, error } => {
+                if let Some(peer_id) = &peer_id {
+                    // Create or extend temporary ban, but only if we are not offline
+                    if let Some(shared) = self.shared_weak.upgrade() {
+                        // One peer is possibly a node peer is connected to, hence expecting more
+                        // than one for online status
+                        if shared.connected_peers_count.load(Ordering::Relaxed) > 1 {
+                            let should_temporary_ban = match &error {
+                                DialError::Transport(addresses) => {
+                                    // Ignoring other errors, those are likely temporary ban errors
```

Comment:
This assumption could create a subtle error in case of another "Other" transport error. Please, consider introducing CustomTransportError enum and matching exactly.

Reply:
This comes from a boxed libp2p transport. There are two enum variants on `TransportError`: `MultiaddrNotSupported(Multiaddr)` and `Other(TErr)`. And `TErr` here is `std::io::Error`, I'm not sure I want to do refactoring across everything just to get a specific nested enum here.

THe only thing we're potentially missing here is that we'll not be temporarily banning peers in some cases, which I think is not the end of the world.

Reply:
Can't we create our own `Into<Error>` impl? 
```
  let error =
      io::Error::new(io::ErrorKind::Other, "Peer is temporarily banned");
  return Err(TransportError::Other(error.into()));
```

Reply:
It has to be `TransportError::Other(std::io::Error)` or we have a big refactoring to do (I think).

I added `T::Error: From<io::Error>` because I knew it was true and it let me keep everything generic in `CustomTransportWrapper` so it still compiles.

Feel free to try making it nicer and submitting PR against this branch, I don't mind it at all. But I found it not super pleasant to deal with.

Reply:
I meant something like this: https://github.com/shamil-gadelshin/subspace/commit/2a64538aef6de7ee9920f2e11df705a56dc9576c

But it seems it is impossible to match anyway.

Reply:
String comparison for errors looks bad but it could create at least some safety: https://github.com/shamil-gadelshin/subspace/commit/d0a3e22357479713a1eca34e20a1f378f6ecb114

Reply:
I don't think there isn't much unsafety here anyway. Banning of unreachable peers is an optimization, it doesn't cause any fundamental unsafety on its own, so I'm inclined to not over-engineer this.